ls1intum/Artemis
# `General`: Fix an issue when sending messages in multi-instance environment

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
Programming exercises did not get scheduled (e.g. for unlocks before exams) and other scheduling-operations could not be executed because of `java.lang.ClassCastException: class [Ljava.lang.Long; cannot be cast to class java.lang.Long` on multi-node environments.

### Description
In this PR (https://github.com/ls1intum/Artemis/pull/5665/files#diff-3d76f824998f2fda8f76f2d3652679ca2a0ca6aa9a226c1ad4b243483d07e72eR139), the type of the payload was changed from `Long` to `Long...` (so thus effectively to `Long[]`), however this change was not reflected in the `InstanceMessageReceiveService` as it still expected a `Long`.

I've added a second method that restores the old behavior for single-argument calls of the method.


### Steps for Testing
**Note:** This change can only be tested with multiple instances, e.g. on the Staging-System (not on the TS0-5)! Additionally, access to the logs is required, so please contact me on Slack so that I can give you the related logs if you test it on the Staging system.
1. Create a programming exercise
2. Change the name of it
3. On node 1, it should show `Received schedule update for programming exercise X`
4. Repeat steps 2-3 several times (~5 times) 

### Review Progress





#### Code Review
#### Manual Tests
